src/Behat/Behat/Transformation/Context/Attribute/TrasnformationAttributeReader.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Transformation\Context\Attribute;
+
+use Behat\Behat\Context\Annotation\DocBlockHelper;
+use Behat\Behat\Context\Attribute\AttributeReader;
+use Behat\Behat\Transformation\Context\Annotation\TransformationAnnotationReader;
+use Behat\Transformation as Attribute;
+use ReflectionMethod;
+
+/**
+ * Reads transformation Attributes from the context class.
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+final class TrasnformationAttributeReader extends TransformationAnnotationReader implements AttributeReader
```

Comment:
Why extending the annotation reader ? Apparently, you are *calling* the public API of the annotation reader as part of your logic, so this could use delegation instead of inheritance.
Or maybe it should not reuse the TransformationAnnotationReader, avoiding to concatenate the pattern just to extract it again (and making it easier to deprecate annotations in the future if their implementation is not a dependency of the attribute reader)

Reply:
My initial implementation was much more tied to the AnnotationReader but you are right that the final implementation does not need inheritance. What I have done is move the common functionality to another utility class. Thanks for the hint